### PR TITLE
[TF FE] Complex type support extended for Separate Bass model.

### PR DESCRIPTION
**Pull request by popovaan**
Description: 
Added support for TensorListConcatV2, with limitation that leading_dims should be empty and shape should be constant.
Fixed a bug in FloorDiv translator.

Added support of complex types for following operations:
- Shape
- Reshape
- Identity
- Squeeze
- StridedSlice
- Transpose

Ticket: 125250

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names
* [x]  Transformation preserves tensor names

Validation:
* [x]  Unit tests
* [x]  Framework operation tests
* [x]  Transformation tests

Documentation:
* [x]  Supported frameworks operations list
* [x]  Guide on how to convert the **public** model
* [x]  User guide update

### CODE REVIEW

**Review by gkrivor — APPROVED** [review]

File: src/frontends/tensorflow_common/src/op/binary_op.cpp [hunk 1]
```diff
@@ -27,6 +30,8 @@
 #include "openvino/op/prelu.hpp"
 #include "openvino/op/squared_difference.hpp"
 #include "openvino/op/subtract.hpp"
+#include "openvino/op/unsqueeze.hpp"
+#include "openvino/opsets/opset13.hpp"
```
Comment (mvafin) [inline src/frontends/tensorflow_common/src/op/binary_op.cpp:34]: Please remove opset include

File: src/frontends/tensorflow_common/src/op/binary_op.cpp [hunk 2]
```diff
@@ -60,6 +73,56 @@ OutputVector translate_binary_op(const NodeContext& node) {
     });
 }
 
+OutputVector translate_mul_op(const NodeContext& node) {
+    default_op_checks(node, 2, {}, true);
+    auto lhs = node.get_input(0);
+    auto rhs = node.get_input(1);
+    auto result = make_shared<v1::Multiply>(lhs, rhs);
+    set_node_name(node.get_name(), result);
```
Comment (mvafin) [inline src/frontends/tensorflow_common/src/op/binary_op.cpp:81]: I suspect this is better to do after `if` resolving complex types

File: src/frontends/tensorflow_common/src/op/squeeze.cpp [hunk 3]
```diff
@@ -5,7 +5,12 @@
 #include "openvino/op/squeeze.hpp"
 
 #include "common_op_table.hpp"
+#include "helper_ops/complex_type_mark.hpp"
 #include "openvino/op/constant.hpp"
+#include "openvino/op/floor_mod.hpp"
+#include "openvino/op/subtract.hpp"
+#include "openvino/opsets/opset8.hpp"
```
Comment (mvafin) [inline src/frontends/tensorflow_common/src/op/squeeze.cpp:12]: Please remove opset include

File: src/frontends/tensorflow_common/src/op/tensor_list_operations.cpp [hunk 4]
```diff
@@ -11,6 +13,7 @@
 #include "openvino/op/convert_like.hpp"
 #include "openvino/op/gather.hpp"
 #include "openvino/op/maximum.hpp"
+#include "openvino/op/multiply.hpp"
```
Comment (mvafin) [inline src/frontends/tensorflow_common/src/op/tensor_list_operations.cpp:16]: Doesn't seem like you use Multiply in this file

**Review by mvafin — CHANGES_REQUESTED** [review]

File: src/frontends/tensorflow_common/src/op/binary_op.cpp (hunk 1 above)
Comment (popovaan) [reply src/frontends/tensorflow_common/src/op/binary_op.cpp]: Done,

File: src/frontends/tensorflow_common/src/op/binary_op.cpp (hunk 2 above)
Comment (popovaan) [reply src/frontends/tensorflow_common/src/op/binary_op.cpp]: Done.

File: src/frontends/tensorflow_common/src/op/squeeze.cpp (hunk 3 above)
Comment (popovaan) [reply src/frontends/tensorflow_common/src/op/squeeze.cpp]: Done.

**Review by mvafin — APPROVED** [review]

**Pull request merged**